### PR TITLE
Fix CSimBulletsyncPacket crash

### PR DESCRIPTION
**Pull request by G-Moris**
Fixed sync weapon fire, which caused the client to crash

Dutchman101, take a look at this please - https://discord.com/channels/278474088903606273/278521065435824128/1246860564030816297

### CODE REVIEW

File: Server/mods/deathmatch/logic/net/CSimBulletsyncPacket.cpp [hunk 1]
```diff
@@ -25,7 +25,11 @@ bool CSimBulletsyncPacket::Read(NetBitStreamInterface& BitStream)
 {
     char cWeaponType;
     BitStream.Read(cWeaponType);
-    m_Cache.weaponType = (eWeaponType)cWeaponType;
+    eWeaponType WeaponType = static_cast<eWeaponType>(cWeaponType);
+    if (!(WeaponType >= WEAPONTYPE_UNARMED && WeaponType < WEAPONTYPE_LAST_WEAPONTYPE))
+        return false;
+
+    m_Cache.weaponType = WeaponType;
```
Comment (botder) [inline Server/mods/deathmatch/logic/net/CSimBulletsyncPacket.cpp]:
```suggestion
    auto weaponType = static_cast<eWeaponType>(cWeaponType);
    if (!(weaponType >= WEAPONTYPE_UNARMED && weaponType < WEAPONTYPE_LAST_WEAPONTYPE))
        return false;

    m_Cache.weaponType = weaponType;
```

**Review by botder — APPROVED** [review]

**Comment by G-Moris** [comment]
botder, please pass my message from the MTA discord to discord to this person - Dutchman101, this is very important for the MTA and our forks, my discord is 'g_moris' (Sorry for the clumsy English)

**Comment by theSarrum** [comment]
https://github.com/multitheftauto/mtasa-blue/blob/ee8bc92907a112a5584844329dbb07cc82326ad1/Server/mods/deathmatch/logic/net/CSimBulletsyncPacket.cpp#L21-L23
Does this mean that the same changes should also be applied to [CBulletsyncPacket.cpp](https://github.com/multitheftauto/mtasa-blue/blob/ee8bc92907a112a5584844329dbb07cc82326ad1/Server/mods/deathmatch/logic/packets/CBulletsyncPacket.cpp#L26)?
___
@G-Moris, don't ask others to pass on your message. He will see it and if he sees fit, he will respond. This is not the place for that.

**Comment by botder** [comment]
Done in a2db617ddd86cb259b016b727ab7e6ad555c3440

**Pull request merged**